sources/subs/InlinePermissionsForm.class.php

```diff
@@ -0,0 +1,245 @@
+
+	private $permissionsObject;
+	private $db;
+
+	/**
+	 * @return string[]
+	 */
+	public function getPermissions()
+	{
+		return $this->permissions;
+	}
+
+	/**
+	 * @param string[] $permissions
+	 */
+	public function setPermissions($permissions)
+	{
+		$this->permissions = $permissions;
+
+		// Load the permission list
+		$this->permissionList = array_map(
+			function ($permission)
+			{
+				return $permission[1];
+			}, $this->permissions
+		);
+
+		// Load the permission settings that guests cannot have
+		$this->illegal_guest_permissions = array_intersect(
+			array_map(
+				function ($permission)
+				{
+					return str_replace(array('_any', '_own'), '', $permission[1]);
+				}, $this->permissions
+			), $this->permissionsObject->getIllegalGuestPermissions()
+		);
+	}
+
+	/**
+	 * @return int[]
+	 */
+	public function getExcludedGroups()
+	{
+		return $this->excluded_groups;
+	}
+
+	/**
+	 * @param int[] $excluded_groups
+	 */
+	public function setExcludedGroups($excluded_groups)
+	{
+		$this->excluded_groups = $excluded_groups;
+	}
+
+	public function __construct()
+	{
+		$this->db = database();
+
+		// Make sure they can't do certain things,
+		// unless they have the right permissions.
+		$this->permissionsObject = new Permissions;
+		$this->illegal_permissions = $this->permissionsObject->getIllegalPermissions();
+	}
+
+	/**
+	 * Save the permissions of a form containing inline permissions.
+	 *
+	 * @param string[] $permissions
+	 */
+	public function save()
+	{
+		global $context;
+
+		$insertRows = array();
+		foreach ($this->permissionList as $permission)
+		{
+			if (!isset($_POST[$permission]))
+				continue;
+
+			foreach ($_POST[$permission] as $id_group => $value)
+			{
+				if (in_array($value, array('on', 'deny')) && !in_array($permission, $this->illegal_permissions))
+					$insertRows[] = array($permission, (int) $id_group, $value == 'on' ? 1 : 0);
+			}
+		}
+
+		// Remove the old permissions...
+		$this->permissionsObject->deletePermissions($this->permissionList);
+
+		// ...and replace them with new ones.
+		require_once(SUBSDIR . '/ManagePermissions.subs.php');
+		replacePermission($insertRows);
+
+		// Do a full child update.
+		$this->permissionsObject->updateChild(array(), -1);
+
+		// Just in case we cached this.
+		updateSettings(array('settings_updated' => time()));
+	}
+
+	/**
+	 * Initialize a form with inline permissions settings.
+	 * It loads a context variables for each permission.
+	 * This function is used by several settings screens to set specific permissions.
+	 *
+	 * @uses ManagePermissions language
+	 * @uses ManagePermissions template
+	 */
+	public function init()
+	{
+		global $context, $txt, $modSettings;
+
+		loadLanguage('ManagePermissions');
+		loadTemplate('ManagePermissions');
+
+		// No permissions? Not a great deal to do here.
+		if (!allowedTo('manage_permissions'))
+			return;
+
+		// Load the permission settings for guests
+		foreach ($this->permissions as $permission)
```

Comment:
Sorry for being a P.I.T.A.
At some point, it was kind of agreed to start using brackets around any loop/condition, even if one-liner.
In these cases with somehow long statements, in particular, it would be nice to start using them. ;)